src/Microsoft.Health.Dicom.Api/Controllers/WorkitemController.Update.cs

```diff
@@ -0,0 +1,77 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Api.Features.Audit;
+using Microsoft.Health.Dicom.Api.Extensions;
+using Microsoft.Health.Dicom.Api.Features.Routing;
+using Microsoft.Health.Dicom.Core.Extensions;
+using Microsoft.Health.Dicom.Core.Features.Audit;
+using Microsoft.Health.Dicom.Core.Messages.Workitem;
+using Microsoft.Health.Dicom.Core.Models;
+using Microsoft.Health.Dicom.Core.Web;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.Health.Dicom.Api.Controllers;
+
+public partial class WorkitemController
+{
+    /// <summary>
+    /// This action requests the update of a UPS Instance on the Origin-Server. It corresponds to the UPS DIMSE N-SET operation.
+    /// </summary>
+    /// <remarks>
+    /// The request body contains all the metadata to be updated in DICOM PS 3.18 JSON metadata.
+    /// </remarks>
+    /// <returns></returns>
+    [HttpPost]
+    [Produces(KnownContentTypes.ApplicationDicomJson)]
+    [Consumes(KnownContentTypes.ApplicationDicomJson)]
+    [ProducesResponseType((int)HttpStatusCode.OK)]
+    [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+    [ProducesResponseType((int)HttpStatusCode.Conflict)]
+    [ProducesResponseType((int)HttpStatusCode.NotAcceptable)]
+    [ProducesResponseType((int)HttpStatusCode.NotFound)]
+    [ProducesResponseType((int)HttpStatusCode.UnsupportedMediaType)]
+    [VersionedPartitionRoute(KnownRoutes.UpdateWorkitemInstancesRoute, Name = KnownRouteNames.PartitionedUpdateWorkitemInstance)]
+    [VersionedRoute(KnownRoutes.UpdateWorkitemInstancesRoute, Name = KnownRouteNames.UpdateWorkitemInstance)]
+    [AuditEventType(AuditEventSubType.UpdateWorkitem)]
+    public async Task<IActionResult> UpdateAsync(string workitemInstanceUid)
+    {
+        // The Transaction UID is passed as the first query parameter 
+        string transactionUid = HttpContext.Request.Query.Keys.FirstOrDefault();
+
+        return await PostUpdateAsync(workitemInstanceUid, transactionUid);
+    }
+
+    private async Task<IActionResult> PostUpdateAsync(string workitemInstanceUid, string transactionUid)
+    {
+        long fileSize = Request.ContentLength ?? 0;
+        _logger.LogInformation("DICOM Web Update Workitem Transaction request received with file size of {FileSize} bytes.", fileSize);
+
+        UpdateWorkitemResponse response = await _mediator.UpdateWorkitemAsync(
```

Comment:
In IHE, it says before updating the workitem, the task performer will update who is updating the workitem. But I didnt see it in the dicom website. We need to make ensure do we need that?

Reply:
Discussed about this already.